packages/node_modules/@webex/webex-core/test/integration/spec/services/services.js

```diff
@@ -476,6 +484,23 @@ describe('webex-core', () => {
 
         assert.notEqual(priorityUrl, nextPriorityUrl);
       });
+
+      it('should reset hosts once all hosts have been marked failed', () => {
+        const priorityServiceUrl = catalog._getUrl(testUrlTemplate.name);
+        const firstPriorityUrl = priorityServiceUrl._getPriorityHostUrl();
+
+        priorityServiceUrl.hosts.forEach(
+          () => {
+            const priorityUrl = priorityServiceUrl._getPriorityHostUrl();
+
+            services.markFailedUrl(priorityUrl);
+          }
+        );
+
+        const lastPriorityUrl = priorityServiceUrl._getPriorityHostUrl();
+
+        assert.equal(firstPriorityUrl, lastPriorityUrl);
+      });
```

Comment:
This test validates that the hosts are reset and are usable after all have been marked as failed.